controller/linodemachine_controller.go

```diff
@@ -446,8 +457,17 @@ func (r *LinodeMachineReconciler) resizeRootDisk(
 
 			return err
 		}
-		diskSize := rootDisk.Size - defaultEtcdDiskSize
-		if err = machineScope.LinodeClient.ResizeInstanceDisk(ctx, linodeInstanceID, rootDiskID, diskSize); err != nil {
+		// dynamically calculate root disk size unless an explcit OS disk is being set
+		additionalDiskSize := 0
+		for _, disk := range machineScope.LinodeMachine.Spec.DataDisks {
+			additionalDiskSize += disk.SizeMB
+		}
+		diskSize := rootDisk.Size - additionalDiskSize
+		if machineScope.LinodeMachine.Spec.OSDisk != nil {
+			diskSize = machineScope.LinodeMachine.Spec.OSDisk.SizeMB
+		}
+
+		if err = machineScope.LinodeClient.ResizeInstanceDisk(ctx, linodeInstanceID, rootDiskID, diskSize); err != nil && !linodego.ErrHasStatus(err, linodeBusyCode) {
```

Comment:
nit: try to not rely on `err` being in the larger scope, when it can be safely scoped to the `if` block

```suggestion
		if err := machineScope.LinodeClient.ResizeInstanceDisk(ctx, linodeInstanceID, rootDiskID, diskSize); err != nil && !linodego.ErrHasStatus(err, linodeBusyCode) {
```

Also, super awesome seeing you use `ErrHasStatus` :grin: